aspnetcore-webapi/UsingResultPatternInNETWebAPI/UsingResultPatternInNETWebAPI/v4/TheResultPattern/CustomResult.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace UsingResultPatternInNETWebAPI.v4.TheResultPattern;
+
+public class CustomResult<T>
+{
+    private readonly T? _value;
+
+    private CustomResult(T value)
+    {
+        Value = value;
+        IsSuccess = true;
+        Error = CustomError.None;
+    }
+
+    private CustomResult(bool isSuccess, CustomError error)
+    {
```

Comment:
Since it is a private method and called only for isSuccess=false, the isSuccess=true condition check is not required.

Reply:
Yes, and no, sure, I can't call it from the outside to create a situation when we've got a contradiction (isSuccess == true && some error). Still, as it is a piece of code that someone else might use and reuse, the fact that we are preventing from creating an inconsistent state seems to be helpful. What do you think?

Reply:
If you want to keep it as is, I don't mind.